internal/pipeline.go

```diff
@@ -63,7 +63,7 @@ func NewPipeline(components []string) (*Pipeline, error) {
 			comp := compInit()
 
 			// request component to parse and validate config of its interest
-			err := comp.Configure()
+			err := comp.Configure(isParent)
```

Comment:
Does isParent makes more sense in terms of name of isDaemon makes it more obvious.

Reply:
that would make it a negative bool cause we only want to run it in the Parent code not the Daemon code

Reply:
Hopefully adding this comment in azstorage where we consume isParent bool helps
// The daemon runs all pipeline Configure code twice. isParent allows us to only validate credentials in parent mode, preventing a second unnecessary REST call.